components/movementsensor/collectors.go

```diff
@@ -19,31 +49,162 @@ func assertMovementSensor(resource interface{}) (MovementSensor, error) {
 	return ms, nil
 }
 
-type lowLevelCollector func(ctx context.Context, ms MovementSensor, extra map[string]interface{}) (interface{}, error)
+func newPositionCollector(resource interface{}, params data.CollectorParams) (data.Collector, error) {
+	ms, err := assertMovementSensor(resource)
+	if err != nil {
+		return nil, err
+	}
+
+	cFunc := data.CaptureFunc(func(ctx context.Context, extra map[string]*anypb.Any) (interface{}, error) {
+		vec, err := ms.LinearVelocity(ctx, data.FromDMExtraMap)
+		if err != nil {
+			// A modular filter component can be created to filter the readings from a component. The error ErrNoCaptureToStore
+			// is used in the datamanager to exclude readings from being captured and stored.
+			if errors.Is(err, data.ErrNoCaptureToStore) {
+				return nil, err
+			}
+			return nil, data.FailedToReadErr(params.ComponentName, position.String(), err)
+		}
+		return pb.GetLinearVelocityResponse{
```

Comment:
Looks like we mixed up the linear velocity collector with the position collector (this function is called `newPositionCollector` but returns a linear velocity collector) 🙂